frontend/public/components/storage-class-form.tsx

```diff
@@ -648,40 +579,61 @@ export class StorageClassForm_ extends React.Component<
       : this.setState({ newStorageClass: newParams });
   };
 
+  addDefaultParams = () => {
+    const defaultParams = this.state.newStorageClass.type
+      ? this.storageTypes[this.state.newStorageClass.type].parameters
+      : {};
+    const hiddenParmas = {};
+    _.each(defaultParams, (values, param) => {
+      if (values.visible && !values.visible() && values.value) {
+        hiddenParmas[param] = values;
+      }
+    });
+    const value = { ...this.state.newStorageClass.parameters, ...hiddenParmas };
+    const newParams = {
+      ...this.state.newStorageClass,
+      parameters: value,
+    };
+
+    return newParams;
+  };
+
   createStorageClass = (e: React.FormEvent<EventTarget>) => {
     e.preventDefault();
 
     this.setState({
       loading: true,
+      error: null,
     });
+    this.setState({ newStorageClass: this.addDefaultParams() }, () => {
+      const { description, type, reclaim } = this.state.newStorageClass;
+      const dataParameters = this.getFormParams();
+      const annotations = description ? { description } : {};
+      const data: StorageClass = {
+        metadata: {
+          name: this.state.newStorageClass.name,
+          annotations,
+        },
+        provisioner: this.storageTypes[type].provisioner,
+        parameters: dataParameters,
+      };
+
+      if (reclaim) {
+        data.reclaimPolicy = reclaim;
+      }
 
-    const { description, type, reclaim } = this.state.newStorageClass;
-    const dataParameters = this.getFormParams();
-    const annotations = description ? { description } : {};
-    const data: StorageClass = {
-      metadata: {
-        name: this.state.newStorageClass.name,
-        annotations,
-      },
-      provisioner: this.storageTypes[type].provisioner,
-      parameters: dataParameters,
-    };
-
-    if (reclaim) {
-      data.reclaimPolicy = reclaim;
-    }
-
-    const volumeBindingMode = _.get(this.storageTypes[type], 'volumeBindingMode', null);
-    if (volumeBindingMode) {
-      data.volumeBindingMode = volumeBindingMode;
-    }
+      const volumeBindingMode = this.storageTypes[type]?.volumeBindingMode;
+      if (volumeBindingMode) {
+        data.volumeBindingMode = volumeBindingMode;
+      }
 
-    k8sCreate(StorageClassModel, data)
-      .then(() => {
-        this.setState({ loading: false });
-        history.push('/k8s/cluster/storageclasses');
-      })
-      .catch((error) => this.setState({ loading: false, error }));
+      k8sCreate(StorageClassModel, data)
+        .then(() => {
+          this.setState({ loading: false });
+          history.push('/k8s/cluster/storageclasses');
```

Comment:
This is not a new thing. But isn't the default side effect of creation is that we move to the details page of the created resource?  @spadgett

Reply:
Yes, that's true, We need to move to the details page if the resource is created.  

I saw mockups in UX call of CNV & storage where there will be a success page and use can go to multiple pages from there but that not implemented right now.